civicrm_entity.module

```diff
@@ -490,6 +491,19 @@ function _civicrm_entity_post_invoke($op, $storage, $entity) {
 }
 
 function civicrm_entity_views_query_alter(ViewExecutable $view, QueryPluginBase $query) {
+  // Provide fully qualified table name in all Views queries
+  $civicrm_connection_name = drupal_valid_test_ua() ? 'civicrm_test' : 'civicrm';
+  $civicrm_connection = Database::getConnection('default', $civicrm_connection_name);
+  $table_queue =& $query->getTableQueue();
+  foreach ($table_queue as $alias => &$table_info) {
+    if (strpos($table_info['table'], 'civicrm_') === 0 && strpos($table_info['table'], '.') === FALSE) {
+      $table_info['table'] = $civicrm_connection->getFullQualifiedTableName($table_info['table']);
+    }
+    if (!empty($table_info['join']->table) && strpos($table_info['join']->table, 'civicrm_') === 0 && strpos($table_info['join']->table, '.') === FALSE) {
+      $table_info['join']->table = $civicrm_connection->getFullQualifiedTableName($table_info['join']->table);
+    }
+  }
```

Comment:
Isn't this doing the exact same thing the `civicrm_views_query` is doing? Why have it live in an alter instead of the query plugin

Reply:
Nothing wrong with the plugin when the base table is a Civi table. But if it won't work for when base table is a Drupal table, then I figured have code only in one place.

Reply:
Turns out the plugin is needed. If commented, with the query alters .. would still get problems with a View with base table civicrm_contact with a relationship to users .. as the bridge relationship would then start to assume the users table was in the Civi database